Index: src/classes/org/jdesktop/wonderland/modules/textchat/client/TextChatPanel.java
===================================================================
--- src/classes/org/jdesktop/wonderland/modules/textchat/client/TextChatPanel.java	(revision 4501)
+++ src/classes/org/jdesktop/wonderland/modules/textchat/client/TextChatPanel.java	(working copy)
@@ -123,7 +123,7 @@
         String date = new SimpleDateFormat("h:mm a").format(new Date());
         String msg = "--- User " + remoteUser +
                 " has joined the world at " + date + " ---\n";
-        messageTextArea.append(msg);
+    //    messageTextArea.append(msg);
         messageTextField.setEnabled(true);
         sendButton.setEnabled(true);
     }
Index: src/classes/org/jdesktop/wonderland/modules/textchat/client/ChatManager.java
===================================================================
--- src/classes/org/jdesktop/wonderland/modules/textchat/client/ChatManager.java	(revision 4501)
+++ src/classes/org/jdesktop/wonderland/modules/textchat/client/ChatManager.java	(working copy)
@@ -314,6 +314,26 @@
         });
     }
 
+    public void showTextChatAll() {
+        SwingUtilities.invokeLater(new Runnable() {
+           public void run() {
+               WeakReference<HUDComponent> ref =
+                       textChatHUDRefMap.get("");
+
+               if(ref == null) {
+                   System.out.println("Text Chat All doesn't exist.");
+                   return;
+               }
+               HUDComponent textChatHUDComponent = ref.get();
+               TextChatPanel chatPanel = textChatPanelRefMap.get(textChatHUDComponent).get();
+               chatPanel.reactivate();
+               textChatHUDComponent.setVisible(true);
+               textChatMenuItem.setState(true);
+
+           }
+        });
+    }
+
     /**
      * @inheritDoc()
      */
